### PR TITLE
feat: integrate user network selection for managing deposit flow for non-NEAR users

### PR DESCRIPTION
**Pull request by jobotics**
- Enhanced network selector to support EVM wallets

### CODE REVIEW

File: src/types/deposit.ts [hunk 1]
```diff
@@ -1,8 +1,14 @@
 import type { SwappableToken } from "../types"
 
+export enum SignInType {
+  NearWalletSelector = "near-wallet-selector",
+  Wagmi = "wagmi",
+}
```
Comment (cawabunga-bytes) [inline src/types/deposit.ts:6]: UserNetworkType { EVM, Near }

File: src/types/deposit.ts [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
+
 export type DepositWidgetProps = {
   tokenList: SwappableToken[]
   userAddress: string | null
+  userNetwork: string | undefined
```
Comment (cawabunga-bytes) [inline src/types/deposit.ts:11]: userNetwork: UserNetworkType

File: src/features/machines/getBalanceMachine.ts [hunk 3]
```diff
@@ -58,7 +59,8 @@ export const getNearNep141Balance = async ({
     const balance = BigInt(parsed)
     return balance
   } catch (err: unknown) {
-    throw new Error("Error fetching balance", { cause: err })
+    console.warn(err, "error fetching near nep141 balance")
+    return 0n
```
Comment (cawabunga-bytes) [inline src/features/machines/getBalanceMachine.ts:63]: Not 0, it's null

File: src/features/deposit/components/DepositForm/index.tsx [hunk 4]
```diff
@@ -424,15 +438,16 @@ function filterBlockchainsOptions(
...
     )
   }
-  return getBlockchainsOptions()
+  return getBlockchainsOptions(userNetwork)
 }
 
 function getDefaultBlockchainOptionValue(
-  token: SwappableToken
+  token: SwappableToken,
+  userNetwork: string
```
Comment (cawabunga-bytes) [inline src/features/deposit/components/DepositForm/index.tsx:446]: userNetwork: UserNetworkType

File: src/features/deposit/components/DepositForm/index.tsx [hunk 5]
```diff
@@ -397,10 +406,15 @@ function getBlockchainsOptions(): Record<
       value: BlockchainEnum.BITCOIN,
     },
   }
+  if (userNetwork !== "near:mainnet") {
+    delete (options as Partial<typeof options>)[BlockchainEnum.NEAR]
```
Comment (cawabunga-bytes) [inline src/features/deposit/components/DepositForm/index.tsx:410]: ??? delete

File: src/features/deposit/components/DepositForm/index.tsx [hunk 6]
```diff
@@ -397,10 +406,15 @@ function getBlockchainsOptions(): Record<
       value: BlockchainEnum.BITCOIN,
     },
   }
+  if (userNetwork !== "near:mainnet") {
```
Comment (cawabunga-bytes) [inline src/features/deposit/components/DepositForm/index.tsx:409]: `"near:mainnet"` magic string

File: src/features/deposit/components/DepositForm/index.tsx [hunk 7]
```diff
@@ -195,78 +201,82 @@ export const DepositForm = () => {
...
-                            token.decimals
-                          )} NEAR`}
-                          <br /> in your wallet.
-                        </TooltipInfo>
-                      )}
-                    <BlockMultiBalances
-                      balance={
-                        token &&
+          {userNetwork === "near:mainnet" &&
+            network === BlockchainEnum.NEAR &&
```
Comment (cawabunga-bytes) [inline src/features/deposit/components/DepositForm/index.tsx:205]: ??

File: src/features/deposit/components/DepositWidget.tsx [hunk 8]
```diff
@@ -23,7 +24,7 @@ export const DepositWidget = ({
           sendTransactionNear={sendTransactionNear}
         >
           <DepositUIMachineFormSyncProvider userAddress={userAddress}>
-            <DepositForm />
+            <DepositForm userNetwork={userNetwork || "unknown"} />
```
Comment (cawabunga-bytes) [inline src/features/deposit/components/DepositWidget.tsx:27]: ?? Unknown?

File: src/features/deposit/components/DepositForm/index.tsx (hunk 5 above)
Comment (jobotics) [reply src/features/deposit/components/DepositForm/index.tsx:410]: We need this to remove redundant option for Non-Near users.

**Review by cawabunga-bytes — APPROVED** [review]

**Pull request merged**